content/en/platform/corda/5.0/deploying-operating/config/secrets.md

```diff
@@ -0,0 +1,92 @@
+* [Default Secrets Service]({{< relref "#default-secrets-service">}})
+* [External Secrets Service]({{< relref "#external-secrets-service--enterprise-icon">}}) {{< enterprise-icon >}}
+
+{{< note >}}
+Any configuration items can be configured as sensitive or not. It is up to you to decide if a particular configuration item should be treated as sensitive.
+{{< /note >}}
+
+## Default Secrets Service
+
+Corda 5 provides a default secrets lookup service. Implementation of this service is in the form of a service that uses symmetric encryption so that the value can be stored encrypted at rest and decrypted with a key derived from a configured salt and passphrase when needed. The salt and passphrase must be specified in the [deployment configuration]({{< relref "../deploying/deploying.md#default-secrets-service" >}}).
+
+For example, the following is a standard configuration:
+
+```
+{
+  "user": "name",
+  "pass": "mypassword"
+}
+```
+
+You can specify the `pass` value as a secret using the `configSecret` value, as follows:
+
+```
+{
+  "user": "name",
+  "pass": {
+    "configSecret": {
+      "encryptedSecret": "<encrypted-password>"
+    } 
+  }
+}
+```
+
+You can use the Corda CLI <a href = "../../reference/corda-cli/secret-config.md">`secret-config` command</a> to generate the configuration for an encrypted value.
+
+You can also use the default secrets lookup service with deployment configurations by using the `configSecret.encryptedSecret` prefix. For example:
+
+```
+-ddatabase.user=db-user
+-ddatabase.pass.configSecret.encryptedSecret=<encryped-db-password>
+-ddatabase.jdbc.url=jdbc:postgresql://db-address:5432/cordacluster
+-ddatabase.jdbc.directory=/opt/corda/drivers
+```
+
+For more information about manually specifying the database deployment configuration, see [Manual Bootstrapping]({{< relref "../deploying/bootstrapping.md#database" >}}).
+
+## External Secrets Service {{< enterprise-icon >}}
+
+In some instances, the default secrets lookup service may not be sufficient. For example, in the case of [Database Connection Configuration]({{< relref "./database-connection.md#configuration-database" >}}), the salt and passphrase used for the encryption would be present in the same set of start-up parameters as the configuration that may be sensitive. This may be adequate if you can ensure that these start-up parameters are sufficiently protected. However, in other cases, it may be preferable to manage these credentials outside Corda.
+
+Corda Enterprise supports integration with [HashiCorp Vault](https://www.vaultproject.io/) as an external secret management system. The URL at which the Vault instance is reachable, the Vault token, and the path to corda created secrets must be specified in the [deployment configuration]({{< relref "../deploying/bootstrapping.md#external-secrets-service" >}}).
```

Comment:
It's probably worth @jamesbr3 reviewing this text as the security team were thinking Vault doesn't provide an improvement in terms of threat model, so maybe the wording of this could be a bit sensitive.

Reply:
ok, so I would probably omit that first paragraph as it draws attention towards a problem, or you could rephrase it slightly and say "as an alternative secrets can be managed using an external secret management system". Do you want to talk about the salt+passphrase as being a problem? probably not, but we have proper recommended option which you should draw attention to.

So write it as "Corda Enterprise supports .....   this is the recommended deployment configuration" ?